tsl/src/bgw_policy/job.c

```diff
@@ -484,6 +575,61 @@ policy_compression_read_and_validate_config(Jsonb *config, PolicyCompressionData
 	}
 }
 
+void
+policy_recompression_read_and_validate_config(Jsonb *config, PolicyCompressionData *policy_data)
+{
+	Oid table_relid = ts_hypertable_id_to_relid(policy_compression_get_hypertable_id(config));
+	Cache *hcache;
+	Hypertable *hypertable =
+		ts_hypertable_cache_get_cache_and_entry(table_relid, CACHE_FLAG_NONE, &hcache);
+	if (policy_data)
+	{
+		policy_data->hypertable = hypertable;
+		policy_data->hcache = hcache;
+	}
+}
+
+bool
+policy_recompression_execute(int32 job_id, Jsonb *config)
+{
+	int32 chunkid;
+	Dimension *dim;
+	PolicyCompressionData policy_data;
+
+	policy_recompression_read_and_validate_config(config, &policy_data);
+	dim = hyperspace_get_open_dimension(policy_data.hypertable->space, 0);
+	chunkid = get_chunk_to_recompress(dim, config);
+
+	if (chunkid == INVALID_CHUNK_ID)
+		elog(NOTICE,
+			 "no chunks for hypertable %s.%s that satisfy recompress chunk policy",
+			 policy_data.hypertable->fd.schema_name.data,
+			 policy_data.hypertable->fd.table_name.data);
+
+	if (chunkid != INVALID_CHUNK_ID)
+	{
+		Chunk *chunk = ts_chunk_get_by_id(chunkid, true);
+		if (hypertable_is_distributed(policy_data.hypertable))
+			policy_invoke_recompress_chunk(chunk);
+		else
+			tsl_recompress_chunk_wrapper(chunk);
+
+		elog(LOG,
+			 "completed recompressing chunk %s.%s",
```

Comment:
Same here as above. Also note that quotes are used elsewhere in this PR for similar errors/message. For example in `tsl_recompress_remote_chunk`.

```suggestion
			 "completed recompressing chunk \"%s.%s\"",
```

Reply:
fixed.